projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3185,7 +3185,8 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      */
     public hideOverlays() {
         this.overlayIDs.forEach(overlayID => {
-            this.overlayService.hide(overlayID);
+            // this.overlayService.hide(overlayID);
+            this.overlayService.detach(overlayID);
             this.overlayService.onClosed.pipe(
```

Comment:
Since `detach` doesn't emit `onClosed`, could this not cause a memory leak?

Reply:
yeah, not sure if this will leak, but absolutely surely we do not need here this subscription
nice catch :)